### PR TITLE
[Branding] Update resource validation procedure.

### LINKED ISSUE #989: [Branding] Remove or allow to disable url validation
This related to #372

[isUrlValid](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.2/src/core/server/rendering/rendering_service.tsx#L317) function rejects some valid values

The validation does not pass if I set relative URLs or embed image into page like url('data:image/svg+xml;base64,.....==')

The validation should be more flexible, or the user must be able to disable validation.

### PR DESCRIPTION
**Pull request by Arbuzov**
New validation procedure allows to embed base64 encoded images

Signed-off-by: Serge Arbuzov <info@whitediver.com>

### Description
This code allow to use base64 encoded values for the branding
 
### Issues Resolved
#989
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by tmarkley — COMMENTED** [review]
Thank you for contributing! This is a great enhancement to custom branding.

I think it would be worthwhile to add some more details to the .yml file about the valid values now that they can be something other than URLs:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/9037b61e9d4da3c44f0ba0fab8f803db5873ef89/config/opensearch_dashboards.yml#L152-L163

Also, can you make the PR title and commit message more specific? "Update resource validation procedure" sells these changes a little short.

Finally, can you add a comment to https://github.com/opensearch-project/documentation-website/issues/241 describing the additional functionality that you're adding here so that we can capture that in the documentation?

File: src/core/server/rendering/rendering_service.test.ts [hunk 1]
```diff
@@ -137,6 +137,29 @@ describe('RenderingService', () => {
     });
   });
 
+  describe('isResourceValid()', () => {
```
Comment (tmarkley) [inline src/core/server/rendering/rendering_service.test.ts:140]: Can you add functional tests to accompany the [existing custom branding functional tests](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/5edf03062d40a92385070f53904b2344c0ffd351/test/functional/apps/visualize/_custom_branding.js#L55)?

**Comment by kavilla** [comment]
Could you update as well: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/rendering/__mocks__/rendering_service.ts.

Without it, I wasn't able to run unit tests. Just to make sure if you could run through the pull request checklist that would be good to ensure verification is there.

Thank you!

**Review by kavilla — CHANGES_REQUESTED** [review]
Unit tests.

**Comment by tmarkley** [comment]
I just realized that this PR was targeting `1.2`. This has to be `main` since we need to merge there first. Please rebase.

**Comment by kavilla** [comment]
> I just realized that this PR was targeting `1.2`. This has to be `main` since we need to merge there first. Please rebase and resolve conflicts.

Another way could be to cherry pick your commit onto `main` and do a force push I believe.

**Comment by tmarkley** [comment]
> > I just realized that this PR was targeting `1.2`. This has to be `main` since we need to merge there first. Please rebase and resolve conflicts.
> 
> Another way could be to cherry pick your commit onto `main` and do a force push I believe.

Yep that works too! Probably easier.

**Comment by tmarkley** [comment]
Hi @Arbuzov, are you able to address the comments above?

**Comment by kavilla** [comment]
Hi @Arbuzov, any updates on the above?

Any thing we can do to help? Thanks!

**Comment by Arbuzov** [comment]
Sorry, new year holydays, then all off this things...
I will manage that this week.

**Comment by kavilla** [comment]
Closing. Please feel free to re-open when ready!

**Pull request closed without merging**